parser.go

```diff
@@ -32,10 +37,21 @@ func NewParser(format string) *Parser {
 		)
 	}
 
-	// Second replace each fileds to regexp grouping
-	quotedFormat := regexp.QuoteMeta(preparedFormat + " ")
-	re := regexp.MustCompile(`\\\$([A-Za-z0-9_]+)(?:\\\$[A-Za-z0-9_])*(\\?([^$\\A-Za-z0-9_]))`).ReplaceAllString(
-		quotedFormat, "(?P<$1>[^$3]*)$2")
+	formatRegex := regexp.MustCompile(`([^$ ]*)\$([a-z_]+)([^$ ]*)([ ]?)`)
+	specialNginxRegexes := getSpecialNginxRegexes()
+	fields := formatRegex.FindAllStringSubmatch(preparedFormat+" ", -1)
+	re := formatRegex.ReplaceAllString(preparedFormat+" ", "$2$4")
+	for _, field := range fields {
+		terminateChar := field[3]
+		if len([]rune(terminateChar)) == 0 {
+			terminateChar = field[4]
+		}
+		if specialRegex, found := specialNginxRegexes[field[2]]; found {
+			re = strings.Replace(re, field[2]+field[4], regexp.QuoteMeta(field[1])+"(?P<"+field[2]+">"+specialRegex+")"+regexp.QuoteMeta(field[3])+field[4], 1)
+		} else {
+			re = strings.Replace(re, field[2]+field[4], regexp.QuoteMeta(field[1])+"(?P<"+field[2]+">[^"+terminateChar+"]*)"+regexp.QuoteMeta(field[3]+field[4]), 1)
+		}
+	}
```

Comment:
For the patch above, please, consider adding more comments around to explain what's going on.